registry/sql-registry/main.py

```diff
@@ -86,6 +86,22 @@ def get_projects_ids() -> dict:
 def get_projects(project: str) -> dict:
     return registry.get_project(project).to_dict()
 
+@router.get("/dependent/{entity}")
+def get_dependent_entities(entity: str) -> list:
+    entity_id = registry.get_entity_id(entity)
+    downstream_entities = registry.get_dependent_entities(entity_id)
+    return list([e.to_dict() for e in downstream_entities])
+
+@router.delete("/entity/{entity}")
+def delete_entity(entity: str):
+    entity_id = registry.get_entity_id(entity)
+    downstream_entities = registry.get_dependent_entities(entity_id)
+    if len(downstream_entities) > 0:
+        raise HTTPException(
+            status_code=500, detail=f"""Entity cannot be deleted as it has downstream/dependent entities.
```

Comment:
Better to use a status code other than 500, my recommendation is "412 precondition failed".

Reply:
updated